src/_nebari/upgrade.py

```diff
@@ -1235,6 +1236,81 @@ def _version_specific_upgrade(
             )
             rich.print("")
 
+        rich.print("\n ⚠️ Upgrade Warning ⚠️")
+
+        text = textwrap.dedent(
+            """
+            Nebari version [green]2024.9.1[/green] introduces changes to how group
```

Comment:
I think this could use a little more detail explaining exactly what's going on here and specifying that the two options here are either all groups or just admin/analyst/developer.  Suggested rewording below (please review for accuracy as well as grammar!)

Nebari version [green]2024.9.1[/green] introduces changes to how group directories are mounted in JupyterLab pods.

In previous versions of Nebari, every Keycloak group in the Nebari realm initiated the creation of a shared file directory that all group members could access in their JupyterLab pod at `~/shared/[group-name]`.

As of Nebari version [green]2024.9.1[/green], only groups that are assigned the `jupyterhub` Client Role `allow-group-directory-creation` will 

You will be asked to confirm whether Nebari should now assign this role to all of your current groups.  If not, only the `admin`, `analyst`, and `developer` group directories will be mounted in JupyterHub pods.

Regardless of your choice now, no data will be lost during this operation. You can re-mount group directories at any time by adding or removing the `allow-group-directory-creation-role` from your  groups in the Keycloak UI.

For more information, please see the [green][link=https://www.nebari.dev/docs/how-tos/group-directory-creation]documentation[/link][/green].